antarest/study/business/model/xpansion_model.py

```diff
@@ -257,3 +258,37 @@ def create_xpansion_candidate(candidate_data: XpansionCandidateCreation) -> Xpan
     candidate = XpansionCandidate.model_validate(candidate_data.model_dump(exclude_none=True))
     validate_xpansion_candidate(candidate)
     return candidate
+
+
+##########################
+# Security criterion part
+##########################
+
+
+class XpansionSecurityPattern(AntaresBaseModel, populate_by_name=True):
+    area: str
+    criterion: float = Field(ge=0)
+
+
+class XpansionSecurityCriterion(AntaresBaseModel, populate_by_name=True, alias_generator=to_camel):
+    stopping_threshold: float = Field(default=1e6, ge=0)
+    criterion_count_threshold: float = Field(default=1, ge=0)
+    patterns: list[XpansionSecurityPattern] = Field(default_factory=list)
+
+
+class XpansionSecurityCriterionUpdate(AntaresBaseModel, populate_by_name=True, alias_generator=to_camel):
```

Comment:
I wonder here if it's worth it to have an "update" feature (and class) and not just a replace (using the class above).
In any case the front  should send the whole object every time.
Plus, the behaviour could be a little misleading, because for the list of patterns we indeed do a replace, not an update.


In general (it's not what we do accross the app though), a PUT is supposed to replace the whole resource, it's more a PATCH that should only update part of the resource.